nemo/collections/common/metrics/classification_accuracy.py

```diff
@@ -0,0 +1,106 @@
+import torch
+from pytorch_lightning.metrics import TensorMetric
+
+__all__ = ['TopKClassificationAccuracy', 'compute_topk_accuracy']
+
+
+def compute_topk_accuracy(correct_counts, total_counts):
+    """
+    Computes the top-k accuracy when provided with a stacked tensor from multiple
+    DDP processes.
+
+    Args:
+        correct_counts: Tensor of shape [D, K], D being the number of processes
+            and K being the top-k parameter.
+        total_counts: Tensor of shape [D, K], D being the number of processes
+            and K being the top-k parameter.
+
+    Returns:
+        A list of length `K`, such that k-th index corresponds to top-k accuracy
+        over all distributed processes.
+    """
+    top_k_scores = []
+
+    for ki in range(correct_counts.shape[-1]):
+        correct_count = correct_counts[:, ki].sum()
+        total_count = total_counts[:, ki].sum()
+        top_k_scores.append(correct_count / float(total_count))
+
+    return top_k_scores
+
+
+class TopKClassificationAccuracy(TensorMetric):
+    """
+    This metric computes numerator and denominator for Overall Accuracy between logits and labels.
+    When doing distributed training/evaluation the result of res=TopKClassificationAccuracy(logits, labels) calls
+    will be all-reduced between all workers using SUM operations.
+    Here contains two numbers res=[correctly_predicted, total_samples]. Accuracy=correctly_predicted/total_samples.
+
+    If used with PytorchLightning LightningModule, include correct_count and total_count inside validation_step results.
+    Then aggregate (sum) then at the end of validation epoch to correctly compute validation WER.
+
+    Example:
+        def validation_step(self, batch, batch_idx):
+            ...
+            correct_count, total_count = self._accuracy(logits, labels)
+            return {'val_loss': loss_value, 'val_correct_count': correct_count, 'val_total_count': total_count}
+
+        def validation_epoch_end(self, outputs):
+            ...
+            correct_count = torch.stack([x['val_correct_count'] for x in outputs]).sum()
+            total_count = torch.stack([x['val_total_count'] for x in outputs]).sum()
+            tensorboard_logs = {'validation_loss': val_loss_mean, 'validation_avg_acc': correct_count / total_count}
+            return {'val_loss': val_loss_mean, 'log': tensorboard_logs}
+
+    Args:
+        top_k: Optional list of integers. Defaults to [1].
+
+    Returns:
+        res: a torch.Tensor object with two elements: [correct_count, total_count]. To correctly compute average
+        accuracy, compute acc=correct_count/total_count
+    """
+
+    def __init__(self, top_k=None):
+        super(TopKClassificationAccuracy, self).__init__(name="TopKClassificationAccuracy")
+
+        if top_k is None:
+            top_k = [1]
+
+        self.top_k = top_k
+
+    def forward(self, logits: torch.Tensor, labels: torch.Tensor) -> torch.Tensor:
+        max_k = max(self.top_k)
```

Comment:
since this is torch.nn.Module.  I wonder if it needs "with torch.no_grad()"?

Reply:
Oh good catch, train graph calls this as well and it wouldn't add gradients to the new nodes

Reply:
Done